Add additional tests

chronitis opened the pull request:
This adds three new test criteria, and corresponding class attributes.
- whether code errors are reported
- a test for code -> result (more general than hello, world on stdout)
- a test for code -> display_data of some MIME type

chronitis committed:
Add tests for error raising, execute and MIME output

chronitis committed:
Update ipython example with these fields

chronitis committed:
Update README with new fields

takluyver commented:
Thanks, I think this looks good. I'd like to make a couple of tweaks to the naming:

```
code_output_samples -> code_execute_result
test_value -> test_execute_result
code_mime_samples -> code_display_data
test_mime -> test_display_data
```

I'd also like to make test_display_data a bit more specific - it should specifically look for display_data messages. This will require a change in the IPython examples, because they're currently generating execute_result messages. Add a call to `IPython.display.display()` around the objects created.

chronitis committed:
output_samples -> execute_result; mime_samples -> display_data

chronitis commented:
I think this addresses the issues. The `display_data` test now expects the first result to be a `display_data` message, which is maybe slightly inflexible, although I'm not sure how the best way to test more complex output which generates both `execute_result` and `display_data`. The simple example in `test_ipython` works correctly in this case.

takluyver commented:
Thanks!
